### PR TITLE
issue #107 Add method KeyRingInfo.isFullyEncrypted()

### LINKED ISSUE #107: Add method KeyRingInfo.isFullyEncrypted()
Similar to existing `KeyRingInfo.isFullyDecrypted()`:
- Should return `true` if all keys in a secret key ring are encrypted.
- Should return `false` for a public key ring

### PR DESCRIPTION
**Pull request by IvanPizhenko**
close #107

### CODE REVIEW

File: pgpainless-core/src/main/java/org/pgpainless/key/info/KeyInfo.java [hunk 1]
```diff
@@ -64,4 +81,24 @@ public static String getCurveName(PGPPublicKey publicKey) {
...
+
+    /**
+     * Returns indication that a secret key is encrypted.
+     *
+     * @param secretKey A secret key to examine.
+     * @return true if secret key is encrypted, false otherwise.
+     */
+    public static boolean isEncrypted(PGPSecretKey secretKey) {
+        return secretKey.getS2KUsage() != 0 && secretKey.getS2K().getType() != S2K.GNU_DUMMY_S2K;
+    }
```
Comment (tomholub) [inline pgpainless-core/src/main/java/org/pgpainless/key/info/KeyInfo.java]: If I get a key:
 - primary key (decrypted)
 - gnu dummy s2k

then:
 - `isFullyDecrypted` should return true.
 - `isFullyEncrypted` should return false

If I get a key:
 - primary key (encrypted)
 - gnu dummy s2k

then:
 - `isFullyDecrypted` should return false.
 - `isFullyEncrypted` should return true

For individual packets, isEncrypted or isDecrypted of a GNU_DUMMY_S2K is not a meaningful question - I would suggest to return null, and handle the case in `isFully...`

Comment (IvanPizhenko) [reply pgpainless-core/src/main/java/org/pgpainless/key/info/KeyInfo.java:93]: @tomholub do you mean we should just ignore subkeys with GNU_DUMMY_S2K? what if primary key has dummy S2K?

Comment (IvanPizhenko) [reply pgpainless-core/src/main/java/org/pgpainless/key/info/KeyInfo.java]: @tomholub so should we just ignore dummy s2k?

Comment (IvanPizhenko) [reply pgpainless-core/src/main/java/org/pgpainless/key/info/KeyInfo.java]: @tomholub Tried to do that in the 3c32bf6

Comment (tomholub) [reply pgpainless-core/src/main/java/org/pgpainless/key/info/KeyInfo.java]: We should not allow dummy s2k to change the output of isFullyEncrypted or isFullyDecrypted - output should purely be based on other non-dummy packets. Best of all would be to skip them entirely somehow.

File: pgpainless-core/src/main/java/org/pgpainless/key/info/KeyRingInfo.java [hunk 2]
```diff
@@ -297,23 +298,44 @@ public boolean isSecretKey() {
...
      *
      * @return true if all secret keys are unencrypted.
      */
     public boolean isFullyDecrypted() {
         if (isSecretKey()) {
             for (PGPSecretKey secretKey : getSecretKeys()) {
-                if (secretKey.getS2KUsage() != 0) {
+                if (!KeyInfo.hasGnuDummyS2K(secretKey) && KeyInfo.isEncrypted(secretKey)) {
                     return false;
                 }
```
Comment (tomholub) [inline pgpainless-core/src/main/java/org/pgpainless/key/info/KeyRingInfo.java]: This looks correct

File: pgpainless-core/src/main/java/org/pgpainless/key/info/KeyRingInfo.java [hunk 3]
```diff
@@ -297,23 +298,44 @@ public boolean isSecretKey() {
...
+     * Sub-keys with S2K of a type GNU_DUMMY_S2K do not affect a result.
+     *
+     * @return true if all secret keys are encrypted.
+     */
+    public boolean isFullyEncrypted() {
+        if (isSecretKey()) {
+            for (PGPSecretKey secretKey : getSecretKeys()) {
+                if (!KeyInfo.hasGnuDummyS2K(secretKey) && KeyInfo.isDecrypted(secretKey)) {
+                    return false;
+                }
```
Comment (tomholub) [inline pgpainless-core/src/main/java/org/pgpainless/key/info/KeyRingInfo.java]: This looks correct too

**Review by tomholub — COMMENTED** [review]
Looks good from reading the highlighted lines

**Comment by IvanPizhenko** [comment]
@vanitasvitae I am getting following CI error on the JDK11
```
FileInfoTest > forYourEyesOnly() FAILED
    org.bouncycastle.openpgp.PGPException at FileInfoTest.java:63
        Caused by: java.security.spec.InvalidKeySpecException at FileInfoTest.java:63
```
while it's ok on JDK 8. I don't think this error should have happened due to my changes, because it's at the following line:
```java
        PGPSecretKeyRing secretKeys = PGPainless.generateKeyRing().simpleEcKeyRing("alice@wonderland.lit");
```

**Comment by vanitasvitae** [comment]
Thats a bug in bouncycastle which I already upstreamed a fix for :)
See #70

**Comment by vanitasvitae** [comment]
Btw: you can run `gradle check` to check if there are checkstyle issues or broken tests which is quicker than waiting for CI :)

**Comment by IvanPizhenko** [comment]
> 
> 
> Thats a bug in bouncycastle which I already upstreamed a fix for :)
> See #70

Great, but it prevents merging this PR.

**Comment by vanitasvitae** [comment]
Looking good. Thanks for the contribution :)

**Pull request merged**